src/Umbraco.Web.UI.Client/src/packages/content/content-type/workspace/views/design/content-type-design-editor.element.ts

```diff
@@ -397,9 +411,16 @@ export class UmbContentTypeDesignEditorElement extends UmbLitElement implements
 
 		const compositionIds = value.selection;
 
-		this.#workspaceContext?.setCompositions(
-			compositionIds.map((unique) => ({ contentType: { unique }, compositionType: CompositionTypeModel.COMPOSITION })),
-		);
+		this.#workspaceContext.setCompositions([
+			...currentOwnerInheritanceCompositions,
+			...compositionIds.map(
+				(unique) =>
+					({
+						contentType: { unique },
+						compositionType: CompositionTypeModel.COMPOSITION,
+					}) as UmbContentTypeCompositionModel,
```

Comment:
Can we avoid type casting here? It usually causes problems if we change the model. Then we won't catch it in this case because the data would always be true.

Reply:
Yeah, god point, but we are creating an object here, do we have anyway to ensure that it is fitting with the type, do we use ´satisfies´ then?